### PR TITLE
Update rpmalloc to the latest version to run on Apple Silicon processors

### LINKED ISSUE #11: Add support for aarch64-apple-darwin
Currently fails building `rpmalloc.c` on inline assembly

### PR DESCRIPTION
**Pull request by BeastLe9enD**
### Checklist

* [x] I have read the [Contributor Guide](../../CONTRIBUTING.md)
* [x] I have read and agree to the [Code of Conduct](../../CODE_OF_CONDUCT.md)
* [x] I have added a description of my changes and why I'd like them included in the section below

### Description of Changes

I updated rpmalloc to the latest version which fixes a compiler error on Apple Silicon processors.

### Related Issues

Fix #11

### CODE REVIEW

**Comment by repi** [comment]
Nice, so this should fix #11?

**Review by repi — CHANGES_REQUESTED** [review]
Thanks!

Can you also update this field with the new rpmalloc version in `Cargo.toml`, it is the version number after the `+`

```
[package]
name = "rpmalloc-sys"
version = "0.2.2+1.4.1"
```

File: rpmalloc-sys/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "rpmalloc-sys"
-version = "0.2.2+1.4.1"
+version = "0.2.2+1.4.3"
```
Comment (repi) [inline rpmalloc-sys/Cargo.toml:3]: Oh wait looked closer and the submodule you updated here seems to not point to the `1.4.3` release of rpmalloc, but to the latest commit https://github.com/mjansson/rpmalloc/commit/b097fd0916500439721a114bb9cd8d14bd998683, and there hasn't been a newer official numbered release yet that includes the Mac fixes from https://github.com/mjansson/rpmalloc/pull/275

We can update and release with an in-development `rpmalloc` submodule, but then the `rpmalloc-sys` version needs to specify the commit instead in the metadata like this:

```suggestion
version = "0.2.2+b097fd0"
```

**Review by repi — CHANGES_REQUESTED** [review]
Oh wait the submodule you updated here seems to not point to the `1.4.3` release of rpmalloc, but to the latest commit https://github.com/mjansson/rpmalloc/commit/b097fd0916500439721a114bb9cd8d14bd998683, and there hasn't been a newer official numbered release yet that includes the Mac fixes from 

We can update and release with an in-development `rpmalloc` submodule, but then the `rpmalloc-sys` version needs to specify the commit instead in the metadata,

File: rpmalloc-sys/Cargo.toml (hunk 1 above)
Comment (BeastLe9enD) [reply rpmalloc-sys/Cargo.toml]: Yes, I actually took the latest version. But I agree with you, taking ```b097fd0``` as the version is more correct!

**Review by repi — APPROVED** [review]

**Comment by repi** [comment]
Released a new v0.2.3 of `rpmalloc-sys` with this fix

**Pull request merged**